aw/ViewControllers/Reaches/Detail/ReachDetailViewController.swift

```diff
@@ -0,0 +1,225 @@
+
+            readMoreButton.isHidden = !descriptionLabel.isTruncated
+        } else {
+            descriptionLabel.text = "Loading"
+        }
+
+        if let photoUrl = reach.photoUrl {
+            imageView.loadFromUrlAsync(urlString: photoUrl)
+        } else {
+            imageViewHeightContraint.constant = 0
+        }
+
+
+        if let difficulty = reach.difficulty {
+            difficultyLabel.text = "Class \(difficulty)"
+        } else {
+            difficultyLabel.text = "N/A"
+        }
+        if let length = reach.length {
+            lengthLabel.text = "\(length) miles"
+        } else {
+            lengthLabel.text = "Unknown"
+        }
+
+        if reach.avgGradient != 0 {
+            gradientLabel.text = "\(reach.avgGradient) fpm"
+        } else {
+            gradientLabel.text = "Unknown"
+        }
+
+        reccomendationLabel.text = reach.runnable
+        reccomendationLabel.textColor = reach.color
+
+        // read more button
+
+
+        if reach.gageId == 0 {
+            seeGageInfoView.isHidden = true
+        }
+    }
+
+    func setupRefreshControl() {
+        let refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action: #selector(refreshReach(sender:)), for: .valueChanged)
+        scrollView.refreshControl = refreshControl
+    }
+
+    @objc func refreshReach(sender: UIRefreshControl?) {
+        guard let reach = reach, let context = managedObjectContext else { return }
+
+        self.scrollView.refreshControl?.beginRefreshing()
+
+        AWApiHelper.updateReachDetail(reachID: String(reach.id), viewContext: context) {
+            self.scrollView.refreshControl?.endRefreshing()
+            self.drawInfo()
+            if let parentVC = self.parent {
+                for vc in parentVC.childViewControllers {
+                    if let mapVC = vc as? ReachDetailMapViewController {
+                        mapVC.reloadAnnotations()
+                    }
+                }
+            }
+        }
+        AWApiHelper.updateReaches(reachIds: [String(reach.id)], viewContext: context) {
+            self.drawInfo()
+        }
+    }
+
+    func share(_ sender: Any?) {
+        guard let reach = reach,
+            let section = reach.section,
+            let name = reach.name,
+            let url = reach.url else { return }
+
+        let title: String
+        if reach.runnable.count > 0 {
+            title = "\( section) of \( name ) is \( reach.runnable )"
+        } else {
+            title = "\( section ) of \( name )"
+        }
+        let activityController = UIActivityViewController(activityItems: [title, url], applicationActivities: nil)
+        if let sender = sender as? UIView {
+            activityController.popoverPresentationController?.sourceView = sender
+        }
+
+        present(activityController, animated: true, completion: nil)
+    }
+}
+
+
+
+extension ReachDetailViewController: RunDetailViewControllerType, MOCViewControllerType {
```

Comment:
Unneeded?

Reply:
I've been doing all protocol compliance on extensions, even though the variables are in the main class to make things cleaner